### PR TITLE
Track subpackages with "python" in the name

### PR DESCRIPTION
**Pull request by encukou**
#  Track all subpackages with a "python" component in the name

Track all subpackages with `python2`, `python3` or `python` as a
dash-separated component of the name.
For subpackages that don't have a Python dependency (as found by
the existing mechanism), mark them `python2` or `python3`
according to the names. (If it's just `python`, don't mark it
either way.)

This will find a lot of new "misnamed requires".

<s>This should cover `python2-networkx` and similar metapackages, so it
fixes: https://github.com/fedora-python/portingdb/issues/459
</s>

 ## Update Fedora data to pick up plugin changes

**missing -> idle (3)**
- check-mk
- python-mtTkinter
- python2-docs

**missing -> released (8)**
- libdnet
- openmpi
- python-basemap-data
- python-repoze-what
- python-rpm-macros
- python-typeshed
- python3-docs
- texlive

**released -> idle (3)**
- python-docker
- python-oslo-config
- python-pyface

**released -> legacy-leaf (1)**
- nautilus-python

**_Naming status changes_**
**missing -> requires misnamed (12)**
- gedit-plugins
- jython
- pypy
- pypy3
- python34
- python35
- python36
- fawkes
- backintime
- python-rpm-macros
- redhat-rpm-config
- texlive

**requires ok -> requires misnamed (51)**
- ibus
- prepaid-manager-applet
- gcc-python-plugin
- gstreamer-rtsp
- libldb
- livecd-tools
- nbdkit
- python-afl
- python-basemap
- python-cartopy
- python-collectd_systemd
- python-formencode
- python-gencpp
- python-genlisp
- python-genpy
- python-ipykernel
- python-metakernel
- root
- python-multilib
- python-music21
- python-oslo-cache
- python-oslo-concurrency
- python-oslo-db
- python-oslo-log
- python-oslo-middleware
- python-oslo-policy
- python-oslo-privsep
- python-oslo-utils
- python-oslo-versionedobjects
- python-oslo-vmware
- python-virtualenv
- python2
- python3
- pyside-tools
- python-qt5
- swatchbooker
- copr-rpmbuild
- ecryptfs-utils
- jpype
- pulp-rpm
- python-aiosmtpd
- python-atpublic
- python-cov-core
- python-flufl-bounce
- python-flufl-i18n
- python-flufl-lock
- python-flufl-testing
- python-iso8601
- python-nose2
- python-ryu
- python-sphinx

### CODE REVIEW

**Comment by encukou** [comment]
@hroncok, what do you think about this approach?

**Comment by hroncok** [comment]
Looking at this.

**Comment by hroncok** [comment]
Should we be worried about the mingw stack?

**Comment by hroncok** [comment]
> This will find a lot of new "misnamed requires".

Taskotron check uses a whitelist:

https://github.com/fedora-python/taskotron-python-versions/blob/16ca63862428ec1012e8ec3f79db977291bd14a1/taskotron_python_versions/two_three.py#L35

Doesn't portingdb use that code? Or is it just that latest release is soo old?

File: dnf-plugins/py3query.py [hunk 1]
```diff
@@ -244,6 +244,21 @@ def run(self):
...
+            'python': 0,
+            'python2': 2,
+            'python3': 3,
+        }.items():
+            for pattern in '{}-*', '*-{}', '*-{}-*':
+                name_glob = pattern.format(name)
+                query = self.pkg_query.filter(name__glob=name_glob)
+                message = 'Getting {} packages'.format(name_glob)
+                for pkg in progressbar(query, message):
+                    if pkg not in python_versions:
```
Comment (hroncok) [inline dnf-plugins/py3query.py]: What about `and not pkg.stratswith('mingw')`?

**Comment by encukou** [comment]
> Taskotron check uses a whitelist. Doesn't portingdb use that code?

Not for this part. Taskotron is used to check py2-only binaries and naming policy.

Taskotron doesn't really apply to this PR. This is about the subset of Fedora's packages that portingdb tracks; Taskotron runs on everything.

**Comment by hroncok** [comment]
I meant for the missnamed requires. Some of them are already tracked in taskotron as "not missnamed". Reusing that would give us less false missnamed requires hits.

**Comment by encukou** [comment]
Rebased onto #461, the latest data update.

Meanwhile, the networkx maintainer dropped all python2 subpackages anyway :/

**Comment by encukou** [comment]
> I meant for the missnamed requires. Some of them are already tracked in taskotron as "not missnamed". Reusing that would give us less false missnamed requires hits.

That's for another issue. I'm not worrying about misnamed packages right now; there's a lot of other stuff going on.

File: dnf-plugins/py3query.py (hunk 1 above)
Comment (hroncok) [reply dnf-plugins/py3query.py]: AFAIK the binary packages are called mingw32- or mingw64-

Comment (hroncok) [reply dnf-plugins/py3query.py]: oh, it's sourcerpm.startswith, nice!

**Pull request merged**